debug_toolbar_user_switcher/views.py

```diff
@@ -32,12 +32,16 @@ def login(request, **kwargs):
     user.backend = settings.AUTHENTICATION_BACKENDS[0]
     auth.login(request, user)
 
-    return HttpResponseRedirect(request.POST.get("next", "/"))
+    return HttpResponseRedirect(
+        request.POST.get("next", request.META.get("HTTP_REFERER", "/")),
```

Comment:
Yes, direct use of the referer here is insecure as an open redirect, but then so was the `next` parameter. As this is a debug tool I think the risk is fairly minimal.